src/i18n.d.ts

```diff
@@ -2,13 +2,21 @@
 // TypeScript Version: 2.8
 
 import { ComponentType, ReactNode } from 'react'
+import { InterpolationOptions } from 'node-polyglot'
 
 interface I18nProps {
   children: ReactNode
   /** Locale to use, e.g. `en` */
-  locale: number
+  locale: string | number
   /** A dictionary of translations */
   messages: object
+
+  allowMissing?: boolean,
+  onMissingKey?: (key: string, options?: InterpolationOptions, locale?: string) => string,
+  interpolation?: {
+    suffix: string,
+    prefix: string
+  },
```

Comment:
What about: 

```
interface InterpolationOptions {
  smart_count?: number | { length: number }
  _?: string

  [interpolationKey: string]: any
}

interface InterpolationTokenOptions {
  prefix?: string
  suffix?: string
}

interface I18nProps {
  children: ReactNode
  /** Locale to use, e.g. `en` */
  locale: number
  /** A dictionary of translations */
  messages: object
  /** control whether missing keys in a t call are allowed */
  allowMissing?: boolean;
  /** called instead of the default functionality if allowMissing */
  onMissingKey?: (key: string, options?: Polyglot.InterpolationOptions, locale?: string) => string
  /** change the substitution syntax for interpolation */
  interpolation?: InterpolationTokenOptions
}
```

Reply:
Thanks for the review.

Are you suggesting we don't use `InterpolationOptions`  from node-polyglot?

I think locale type should be `string | number` since even the example is using 'en'. Will make this change and push again

Reply:
I absolutely agree with you about `string | number` for locale option

`InterpolationOptions` could be used from node-polyglot types but is I thought that DefinitelyTyped packages usually took some time to follow updates. Perhaps it would have been easier to directly control all types of the API internally. But I agree, maintaining the node-polyglot types in parallel to maintain this library is sooo much cleaner

Reply:
Oh and maybe we could add `pluralRules` options too? This option is also not typed on DefinitelyTyped for example :(

Reply:
Totally agree on the point to contribute to typings for `node-polyglot` and use the types from there 👍